src/main/java/uk/gov/hmcts/divorce/systemupdate/service/task/GenerateConditionalOrderPronouncedDocument.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmcts.divorce.systemupdate.service.task;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.type.ListValue;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.task.CaseTask;
+import uk.gov.hmcts.divorce.document.CaseDataDocumentService;
+import uk.gov.hmcts.divorce.document.content.ConditionalOrderPronouncedTemplateContent;
+import uk.gov.hmcts.divorce.document.model.DivorceDocument;
+
+import java.util.Optional;
+
+import static uk.gov.hmcts.divorce.document.DocumentConstants.CONDITIONAL_ORDER_PRONOUNCED_DOCUMENT_NAME;
+import static uk.gov.hmcts.divorce.document.DocumentConstants.CONDITIONAL_ORDER_PRONOUNCED_TEMPLATE_ID;
+import static uk.gov.hmcts.divorce.document.model.DocumentType.CONDITIONAL_ORDER_GRANTED;
+
+@Component
+@Slf4j
+public class GenerateConditionalOrderPronouncedDocument implements CaseTask {
+
+    @Autowired
+    private CaseDataDocumentService caseDataDocumentService;
+
+    @Autowired
+    private ConditionalOrderPronouncedTemplateContent conditionalOrderPronouncedTemplateContent;
+
+    @Override
+    public CaseDetails<CaseData, State> apply(CaseDetails<CaseData, State> caseDetails) {
+
+        final Long caseId = caseDetails.getId();
+        final CaseData caseData = caseDetails.getData();
+
+        log.info("Generating Conditional Order pronounced pdf for CaseID: {}", caseDetails.getId());
+
+        caseDataDocumentService.renderDocumentAndUpdateCaseData(
```

Comment:
I think we will need to check if document was already generated and added to the case because if the bulk case event fails and caseworker retries this then we will add multiple documents?

Reply:
@nitinprabhu reverted the doc store removal logic.